lib/ecto/uuid.ex

```diff
@@ -29,9 +29,9 @@ defmodule Ecto.UUID do
   catch
     :error -> :error
   else
-    casted ->
-      {:ok, casted}
+    casted -> {:ok, casted}
   end
+  def cast(<< _::128 >> = binary), do: binary |> encode() |> cast()
```

Comment:
Can we introduce those checks to `encode/1`? This would also mean we'd need to change encode to return `{:ok, string} | :error`, but that shouldn't be a problem since it's a private function.